Add migrations

Hwesta opened the pull request:
refs #6466

Enable South, and update version.  Add migrations for 0.2.2 and 0.3.0 for locations and administration.

Hwesta committed:
Add migrations

refs #6466

Enable South, and update version.  Add migrations for 0.2.2 and 0.3.0
for locations and administration.

mistydemeo commented:
Whoa, nice! Looks good. This is working well in testing?

Will this work

a) For fresh installs;
b) For installs from previous stable storage-service releases?

jhsimpson committed:
Adding uninstall script and supporting proper updating

Refs #6359 #6466

Ubuntu package will now work when updating from 0.2.2 to 0.3.0.
Will also work when uninstalling 0.3.0

jhsimpson commented:
I added a commit that makes this work with the ubuntu packages.  I tested with a fresh install and with an update from 0.2.2 to 0.3.0.  This is ready to be merged into qa/0.x branch.

Hwesta commented on debian/postrm:
```diff
@@ -0,0 +1,18 @@
+
+#clean up storage service directories
+rm -rf /usr/share/python/archivematica-storage-service
+rm -rf /var/archivematica/storage*
```
This will remove pointer files stored in the SS internal storage - do we want that?

jhsimpson replied:
I think so yes.  This is run when the user is un-installing the storage service.  We are not supporting an interactive un-install - I think it is fair to say don't un-install unless you have backed up whatever data you want to keep.

mistydemeo replied:
I think Debian packages don't typically remove content from `/var`, but we stomp all over the FHS already so it's not really a big deal.

jhsimpson replied:
Agreed, Archivematica and the Storage Service are not FHS compliant.  Archivematica puts the shared Directory structure used in processing in /var/archivematica, and the storage service followed this convention by putting its internal data in /var/archivematica/storage-service.

Both should probably be in /var/lib/, not /var.  It seems to be okay according to both FHS and debian/ubuntu practice for a package to delete items from its own /var/lib/<package-name> directory.

mistydemeo replied:
Issue for discussing FHS compliance here: https://projects.artefactual.com/issues/6578

Hwesta commented:
Looks good.
